### DISCUSSION TITLE
model metadata

### DISCUSSION THREAD

**Discussion by ddbuscombe** [Doodleverse/segmentation_gym#44] [Ideas]
we need better ways to keep track of individual trained models ... what data they were used on, what hyperparameters, etc

[option 1] a bespoke option 
--> config
--> history
--> metrics
json files that can be appended, then parsed

[option 2] conform to a standard e.g. W&B or [MLFlow](https://mlflow.org/docs/latest/tracking.html#tensorflow-and-keras)

what it needs to contain
- [ ] store info on what training data was used to train a model
- [ ] use unique IDs for trained model weights
- [ ] store all values from config file with model weights?
- [ ] store metrics?
- [x] during implementation, keep track of model implementation choices

what type of database?

**Comment by ebgoldstein** [comment]
is this like MLFlow or Weights & Biases?

**Comment by ddbuscombe** [comment]
could be

**Comment by ddbuscombe** [comment]
First implementation in bf1962f

**Comment by ddbuscombe** [comment]
Implement Hugginface-style model cards see  #41
